virttest/utils_test/qemu.py

```diff
@@ -1461,27 +1461,131 @@ def action_after_suspend(self, **args):
         pass
 
 
-class MemoryHotplugTest(object):
+class MemoryBaseTest(object):
 
-    UNIT = "M"
+    """
+    Base class for memory functions.
+    """
+
+    UNIT = "MB"
 
     def __init__(self, test, params, env):
         self.test = test
-        self.env = env
         self.params = params
+        self.env = env
         self.sessions = {}
 
+    def get_vm_mem(self, vm):
+        """
+        Get memory assigned to VM;
+        """
+        PC_DIMM = qdevices.Dimm
+        mem_str = vm.params.get("mem")
+        size_strs = ['%sMB' % mem_str] if mem_str.isdigit() else [mem_str]
+        pc_dimms = [_ for _ in vm.devices if isinstance(_, PC_DIMM)]
+        mem_ids = map(lambda x: x.get_param('memdev'), pc_dimms)
+        mem_devs = map(lambda x: vm.devices.get_by_id(x)[0], mem_ids)
+        size_strs += map(lambda x: x.get_param('size'), mem_devs)
+        total_mem = sum(map(lambda x: self.normalize_mem_size(x), size_strs))
+        mem_str = "%s %s" % (total_mem, self.UNIT)
+        logging.info("Assigned %s memory to '%s'" % (mem_str, vm.name))
+        return total_mem
+
     @classmethod
-    def normalize_mem_mb(cls, str_size):
+    def normalize_mem_size(cls, str_size):
         """
         Convert memory size unit
         """
         args = ("%sB" % str_size, cls.UNIT, 1024)
```

Comment:
no, str_size many be have not a unit that mean we that a size string with Byte, we convert the value to  MB, else if it has a unit 'B' will be append to the end like "KB", "GB" or "TB" so that will not impact value convert.  if origin size string is "KB", "MB" or "GB", origin string will be update to "KBB", "MBB" or "GBB" it will also not impact covert string size.